components/engine_rocks/src/db_vector.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_traits::DBVector;
+use rocksdb::DBVector as RawDBVector;
+use std::fmt::{self, Debug, Formatter};
+use std::ops::Deref;
+
+pub struct RocksDBVector(RawDBVector);
+
+impl RocksDBVector {
```

Comment:
I'm wondering why we need to wrap `RawDBVector` with `RocksDBVector`? why not use `rocksdb::DBVector` directly?

Reply:
Without wrapping it it can't implement the DBVector trait.

Reply:
but `RawDBVector` does implement Debug + Deref<Target = [u8]> + for<'a> PartialEq<&'a [u8]> {}

Reply:
Yes. It could be made to automatically implement DBVector with a blanket implementation. It's not obvious to me that that is a better solution than the newtype written here. The implementation here has the advantage that it follows a pattern that is consistent with the rest of the engine_rocks crate: defining the types that implement the traits from engine_traits. Introducing a new pattern so that an out-of-crate type can implement an engine trait increases the cognitive burden of maintaining the crate.

Reply:
got it